Games/Solitaire/SolitaireWidget.h

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2020, Till Mayer <till.mayer@web.de>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+#include "CardStack.h"
+#include <LibGUI/Painter.h>
+#include <LibGUI/Widget.h>
+
+#define WIDTH 640
+#define HEIGHT 480
+
+class SolitaireWidget final : public GUI::Widget {
+    C_OBJECT(SolitaireWidget)
+public:
+    virtual ~SolitaireWidget() override;
+    void setup();
+
+private:
+    SolitaireWidget(NonnullRefPtr<GUI::Window>&, Function<void(uint32_t)>&& on_score_update);
+
+    class Animation {
+    public:
+        Animation() {};
+        Animation(RefPtr<Card> animation_card, float gravity, int x_vel, float bouncyness)
+            : m_animation_card(animation_card)
+            , m_gravity(gravity)
+            , m_x_velocity(x_vel)
+            , m_bouncyness(bouncyness) {};
+
+        RefPtr<Card> card() { return m_animation_card; }
```

Comment:
Since ownership is not transferred here, we can just return a `const Card*` perhaps?
Or if there is always guaranteed to be a `Card` in all `Animation` objects, we can return `const Card&` (and adjust the smart pointer types as well.)